linode/manager
# change: [M3-8415] - Hide SMTP warning for Linodes and accounts that have SMTP enabled

## Description 📝
Leverages the new `"SMTP Enabled"` account & Linode capabilities to hide the SMTP warning when it doesn't apply.

The SMTP warning appears in two places:

- At the bottom of the Linode create flow. This warning is now hidden for accounts with the new `"SMTP Enabled"` account capability.
- In the Linode details page as a dismissible banner. This warning is now hidden when the `"SMTP Enabled"` Linode capability is present on the selected Linode.

## How to test 🧪

### Prerequisites
Use the alpha API environment.

You will need an account with at least one Linode.

In Admin: remove the `SMTP_ENABLED` customer tag from the account or ensure that it's not present. For the test Linode, check that the *Net Filter Exceptions* field appears as below or change it to match:

![Screenshot 2024-09-23 at 4 56 53 PM](https://github.com/user-attachments/assets/68c03927-6c7f-47f8-a7c0-5f8e8f16974b)

### Testing steps
1. Verify the SMTP banner appears in both locations. If it's not appearing for the Linode, you may have dismissed it. Reset your preferences using the Preference Editor: [http://localhost:3000/profile/settings?preferenceEditor=true](http://localhost:3000/profile/settings?preferenceEditor=true).
2. Enable SMTP for a particular Linode by removing the SMTP Net Filter Exceptions.
3. Verify the SMTP warning banner no longer appears for that particular Linode. However, it should still appear for other Linodes and at the bottom of the Create Linode flow
4. Add the `SMTP_ENABLED` customer tag to your account
5. Verify the SMTP warning no longer appears in the Linode Create flow

## As an Author I have considered 🤔

*Check all that apply*
